code/modules/admin/holder2.dm

```diff
@@ -49,25 +49,24 @@ var/list/admin_datums = list()
 
 /*
 checks if usr is an admin with at least ONE of the flags in rights_required. (Note, they don't need all the flags)
-if rights_required == 0, then it simply checks if they are an admin.
+if rights_required == (), //an empty list// then it simply checks if they are an admin.
 if it doesn't return 1 and show_msg=1 it will prints a message explaining why the check has failed
 generally it would be used like so:
 
 proc/admin_proc()
-	if(!check_rights(R_ADMIN)) return
+	if(!check_rights(list(R_ADMIN))) return
 	world << "you have enough rights!"
 
 NOTE: It checks usr by default. Supply the "user" argument if you wish to check for a specific mob.
 */
-/proc/check_rights(rights_required, show_msg=1, var/mob/user = usr)
+/proc/check_rights(var/list/rights_required = list(), show_msg=1, var/mob/user = usr)
 	if(user && user.client)
-		if(rights_required)
+		if(rights_required.len)
 			if(user.client.holder)
-				if(rights_required & user.client.holder.rights)
+				for(var/required in rights_required)

```

Comment:
Missing check here. This will _always_ return true as it is presently written.
Needs to be this:
```
for(var/required in rights_required)
 if(required in user.client.holder.rights)
  return 1
```